src/main/java/uk/gov/digital/ho/pttg/application/HmrcClient.java

```diff
@@ -234,39 +241,66 @@ private String stripPlaceholderQueryParams(String href) {
     private String getIndividualLink(Individual individual, String accessToken, String matchUrl) {
         log.info("POST to {}", matchUrl);
 
-        Resource<String> resource = restTemplate.exchange(URI.create(matchUrl), HttpMethod.POST, createEntity(individual, accessToken), linksResourceTypeRef).getBody();
+        Resource<String> resource = null;
+        List<String> candidateNames = generateCandidates(individual.getFirstName(), individual.getLastName());
+
+        int retries = 0;
+        boolean success = false;
+
+        while (!success && retries < candidateNames.size()) {
+            try {
+                String[] names = candidateNames.get(retries).split("\\s+");
+                individual.setFirstName(names[0]);
+                individual.setLastName(names[1]);
+                resource = restTemplate.exchange(URI.create(matchUrl), HttpMethod.POST, createEntity(individual, accessToken), linksResourceTypeRef).getBody();
+                success = true;
+            } catch (HttpClientErrorException ex) {
+                if (ex.getStatusCode().equals(FORBIDDEN)) {
+                    retries++;
+                } else if (ex.getStatusCode().equals(UNAUTHORIZED)) {
+                    throw new HmrcUnauthorisedException(ex.getMessage(), ex);
+                } else {
+                    throw ex;
+                }
+            }
+        }
+
+        if (retries == candidateNames.size()) {
+            throw new HmrcNotFoundException(String.format("Unable to match: %s", individual));
```

Comment:
If we refactor this method to return just the individual and create another method to get the link then this could be simplified and the Individual can got back to being immutable. You could also remove the success flag and then return if you find a match or throw the not found if not.